catalog/ui/catalog-ui-search/src/main/webapp/component/query-annotations/query-annotations.view.js

```diff
@@ -0,0 +1,225 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define, setTimeout*/
+define([
+    'jquery',
+    'backbone',
+    'marionette',
+    'underscore',
+    'properties',
+    'js/model/Metacard',
+    'wreqr',
+    './query-annotations.hbs',
+    'maptype',
+    'js/store',
+    'js/CustomElements',
+    'component/property/property.view',
+    'component/property/property',
+    'component/annotation/annotation.collection',
+    'component/annotation/annotation.collection.view',
+    'component/input/textarea/input-textarea.view',
+    'component/loading-companion/loading-companion.view',
+    'component/announcement'
+], function ($, Backbone, Marionette, _, properties, MetaCard, wreqr, template, maptype,
+             store, CustomElements, PropertyView, Property, annotationCollection, annotationCollectionView,
+             TextAreaView, LoadingCompanionView, announcement) {
+
+    return Marionette.LayoutView.extend({
+        setDefaultModel: function () {
+            this.model = this.selectionInterface.getSelectedResults().first();
+        },
+        template: template,
+        tagName: CustomElements.register('query-annotations'),
+        events: {
+            'click .add-annotation-btn ': 'handleCreate',
+            'click .refresh-btn ': 'handleRefresh'
+        },
+        childEvents: {
+            "annotation:add": "handleChildDelete"
+        },
+        regions: {
+            addannotationField: '.add-annotation-field',
+            annotationsList: '.annotations-list'
+        },
+        _annotationsCollection: undefined,
+        _annotations: undefined,
+        initialize: function (options) {
+            this.selectionInterface = options.selectionInterface || this.selectionInterface;
+            if (!options.model) {
+                this.setDefaultModel();
+            }
+        },
+        onBeforeShow: function(){
+            this.clearannotations();
+            this.showAddannotationView();
+            this.getannotationsForQuery();
+            this.showannotationsListView();
+            this.turnOnEditing();
+            this.setupListeners();
+        },
+        clearannotations: function () {
+            if (!this._annotationsCollection) {
+                this._annotationsCollection = new annotationCollection();
+            }
+            this._annotationsCollection.reset();
+        },
+        showAddannotationView: function() {
```

Comment:
✏️  camelCase involving the word `annotation` throughout this file is a bit funky

Reply:
cleaned up the naming